src/index.ts

```diff
@@ -1075,8 +1160,11 @@ class TypeormJsonQuery<TEntity = ObjectLiteral> {
     conditions.forEach((condition) => queryBuilder.andWhere(condition));
     groupByAttr.forEach((field) => queryBuilder.addGroupBy(field));
 
-    if (select.length) {
-      queryBuilder.select(select);
+    if (selectAttributes.length) {
+      // Build distinct only if enabled distinct and distinct attributes exists
+      distinctType !== DistinctType.DISABLED && isDistinctAttributesExist
```

Comment:
Use if block

Reply:
I'll create separate method for avoid nested if block